drivers/flash/flash_native_posix.c

```diff
@@ -0,0 +1,226 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_native_posix);
+
+static const char default_flash_path[] = "flash.bin";
+
+struct flash_native_posix_data {
+	struct k_sem mutex;
+	const char *flash_path;
+	int fd;
+	u8_t *flash;
+};
+
+struct flash_native_posix_config {
+	size_t flash_size;
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+	struct flash_pages_layout layout;
+#endif
+};
+
+#define DEV_NAME(dev) ((dev)->config->name)
+#define DEV_CONFIG(dev) ((dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct flash_native_posix_data *const)(dev)->driver_data)
+
+static int flash_native_posix_read(struct device *dev, off_t offset, void *data,
+				   size_t size)
+{
+	struct flash_native_posix_data *const dev_data = DEV_DATA(dev);
+	const struct flash_native_posix_config *config = DEV_CONFIG(dev);
+
+	if (dev_data->flash == MAP_FAILED) {
+		LOG_ERR("No flash device mapped");
+		return -EIO;
+	}
+
+	if ((offset + size) > config->flash_size) {
+		LOG_WRN("Reading outside of flash boundaries");
+		return -EINVAL;
+	}
+
+	memcpy(data, dev_data->flash + offset, size);
+
+	return 0;
+}
+
+static int flash_native_posix_write(struct device *dev, off_t offset,
+				    const void *data, size_t size)
+{
+	struct flash_native_posix_data *const dev_data = DEV_DATA(dev);
+	const struct flash_native_posix_config *config = DEV_CONFIG(dev);
+
+	if (dev_data->flash == MAP_FAILED) {
+		LOG_ERR("No flash device mapped");
+		return -EIO;
+	}
+
+	if ((offset + size) > config->flash_size) {
+		LOG_WRN("Writing outside of flash boundaries");
+		return -EINVAL;
+	}
+
+	memcpy(dev_data->flash + offset, data, size);
+
+	return 0;
+}
+
+static int flash_native_posix_erase(struct device *dev, off_t offset,
+				    size_t size)
+{
+	struct flash_native_posix_data *const dev_data = DEV_DATA(dev);
+	const struct flash_native_posix_config *config = DEV_CONFIG(dev);
+
+	if (dev_data->flash == MAP_FAILED) {
+		LOG_ERR("No flash device mapped");
+		return -EIO;
+	}
+
+	if ((offset + size) > config->flash_size) {
+		LOG_WRN("Erasing outside of flash boundaries");
+		return -EINVAL;
+	}
+
+	memset(dev_data->flash + offset, 0xff, size);
+
+	return 0;
+}
+
+static int flash_native_posix_write_protection(struct device *dev, bool enable)
+{
+	return 0;
+}
+
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+static void
+flash_native_posix_pages_layout(struct device *dev,
+				const struct flash_pages_layout **layout,
+				size_t *layout_size)
+{
+	const struct flash_native_posix_config *config = DEV_CONFIG(dev);
+	*layout = &config->layout;
+	*layout_size = 1;
+}
+#endif /* CONFIG_FLASH_PAGE_LAYOUT */
+
+static int flash_native_posix_init(struct device *dev)
+{
+	struct flash_native_posix_data *const data = DEV_DATA(dev);
+	const struct flash_native_posix_config *config = DEV_CONFIG(dev);
+
+	k_sem_init(&data->mutex, 1, 1);
+
+	if (data->flash_path == NULL) {
+		data->flash_path = default_flash_path;
+	}
+
+	data->fd = open(data->flash_path, O_RDWR | O_CREAT, (mode_t)0600);
```

Comment:
This is a bit more flimsy than a user may understand from the documentation. If for example I give `-flash=banana/patata.bin` I get a silent segfault (if banana doesn't exist, that is)
Should the documentation mention instead of just "location", something like "location in an existing folder"?

Reply:
> if the user does not enable logging the error will go unnoticed.

Given that it is a POSIX arch driver you could throw a `posix_print_warning` (which always goes to stderr)